docs/advanced/Testing.md

```diff
@@ -1,9 +1,35 @@
 # Examples of Testing Sagas
 
-**WIP**
+There is ongoing discussion in [this issue](https://github.com/redux-saga/redux-saga/issues/518) about the best way to test Sagas.
 
-See Repository Examples:
+
+In the repo we also have two examples showing how Sagas can be tested:
 
 https://github.com/redux-saga/redux-saga/blob/master/examples/counter/test/sagas.js
 
 https://github.com/redux-saga/redux-saga/blob/master/examples/shopping-cart/test/sagas.js
+
+
+
+***External Resources***
+
+Redux Saga Test Plan - https://github.com/jfairbank/redux-saga-test-plan
+
+From the README:
```

Comment:
> In fact, at my work we've come to the firm conclusion that unit testing sagas offers no benefit. The tests are tautological.

I think this is usually true, though it's helpful to test sagas that have branches. That said, I wonder if reducing the tautology would help. 

If we think about unit testing regular functions that have dependencies, you have spies and stubs available to you. You use spies for testing side effects and stubs for testing reads from external dependencies. 

The redux-saga test frameworks force a strictness upon you that normal unit tests with spies and stubs do not have. You must be exact in your ordering and explicit in every spy expectation and stub response. 

If the default behavior were more forgiving, that is, effects were simply recorded and all things were assumed to yield `undefined`, then perhaps the testing could be less tautological and more focused on what you want to test. For example, you could have a test that asserted that when a response was successful an action was put. Nevermind the fact that 3 other things may have happened, your test just tests that one thing, so you set up your "stubs" (yield results) and you assert on your "spies" (`expect(saga.put).toHaveBeenCalledWith(...)`).